### PR TITLE
Remove Spring Boot dependency management

### PR DESCRIPTION
**Pull request by cuijulian**
Depends on https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1086

### CODE REVIEW

File: employee-rostering-backend/src/test/java/org/optaweb/employeerostering/contract/ContractServiceTest.java [hunk 1]
```diff
@@ -141,7 +141,7 @@ public void deleteContractTest() throws Exception {
                             .accept(MediaType.APPLICATION_JSON))
                 .andDo(mvcResult -> logger.info(mvcResult.toString()))
                 .andExpect(status().isOk())
-                .andExpect(content().contentType("application/json;charset=UTF-8"))
+                .andExpect(content().contentType("application/json"))
```
Comment (rsynek) [inline employee-rostering-backend/src/test/java/org/optaweb/employeerostering/contract/ContractServiceTest.java]: What caused the omission of charset in content type?

Comment (cuijulian) [reply employee-rostering-backend/src/test/java/org/optaweb/employeerostering/contract/ContractServiceTest.java]: This was to fix the failing tests. Bumping to the latest Spring version resulted in a different expected content type, like in [this GitHub issue](https://github.com/spring-projects/spring-framework/issues/19041). Omitting the charset fixes this issue.

Comment (rsynek) [reply employee-rostering-backend/src/test/java/org/optaweb/employeerostering/contract/ContractServiceTest.java]: Question is, can we afford omitting the charset? If the charset is not sent via HTTP header, how is some user's client code supposed to know how the JSON is encoded?

Comment (cuijulian) [reply employee-rostering-backend/src/test/java/org/optaweb/employeerostering/contract/ContractServiceTest.java]: @rsynek Good point. I have an alternative that also passes the tests:

`.andExpect(content().contentTypeCompatibleWith("application/json;charset=UTF-8"))`

This way we keep the charset. What do you think of this alternative?

Comment (rsynek) [reply employee-rostering-backend/src/test/java/org/optaweb/employeerostering/contract/ContractServiceTest.java]: Depends how the method works. Does the `contentTypeCompatibleWith` just relax the matching criteria to make it pass with content being encoded via UTF-8 (without the charset being sent via header)?

Comment (cuijulian) [reply employee-rostering-backend/src/test/java/org/optaweb/employeerostering/contract/ContractServiceTest.java]: Hm, according to the [docs](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/test/web/servlet/result/ContentResultMatchers.html#contentTypeCompatibleWith-org.springframework.http.MediaType-) that seems to be the case. Unfortunately after the Spring api changes I don't see a better alternative. I'll see if there is another way to assert contentType. Do you have any recommendations?

Comment (cuijulian) [reply employee-rostering-backend/src/test/java/org/optaweb/employeerostering/contract/ContractServiceTest.java]: @rsynek As this PR is part of a very time-sensitive group of PR's, would it be ok to merge this PR for now, and we can discuss the `contentType` tests later? I can create a Jira for this. Wdyt?

Comment (yurloc) [reply employee-rostering-backend/src/test/java/org/optaweb/employeerostering/contract/ContractServiceTest.java]: @rsynek Valid concern but let's merge without the charset. It's been deprecated. See https://github.com/spring-projects/spring-framework/issues/22788 for explanation.

**Comment by rsynek** [comment]
Travis fails due to its inability to build related PRs to other repositories; let's ignore it.

**Pull request merged**